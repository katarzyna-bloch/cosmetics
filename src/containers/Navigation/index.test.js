import { mount, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Navigation } from '.'

jest.mock('../../components/Navigation', () => {
  const Nav = () => null
  return Nav
})

it('should render correctly', () => {
  const props = {
    fetchCategories: jest.fn(),
    categories: [
      { id: "1", type: "homeSpa", url: "/categories/home-spa", name: "Domowe SPA", description: "Profesjonalne SPA w domowym zaciszu to relaks dla ciała i duszy" },
      { id: "2", type: "cleaning", url: "/categories/cleaning", name: "Oczyszczanie", description: "Oczyszczanie to podstawowy krok w dojściu do pięknej cery" },
      { id: "3", type: "care", url: "/categories/care", name: "Pielęgnacja", description: "Należy przestrzegać codziennej pielęgnacji twarzy" },
    ]    
  }
  const wrapped = mount(<Navigation {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
  wrapped.update()
  expect(props.fetchCategories).toHaveBeenCalled()
})

it('should render correctly without categories', () => {
  const props = {
    fetchCategories: jest.fn(),
    categories: null,
  }
  const wrapped = shallow(<Navigation {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
})
