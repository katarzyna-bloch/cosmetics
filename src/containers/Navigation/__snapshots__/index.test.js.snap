// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Navigation
  categories={
    Array [
      Object {
        "description": "Profesjonalne SPA w domowym zaciszu to relaks dla ciała i duszy",
        "id": "1",
        "name": "Domowe SPA",
        "type": "homeSpa",
        "url": "/categories/home-spa",
      },
      Object {
        "description": "Oczyszczanie to podstawowy krok w dojściu do pięknej cery",
        "id": "2",
        "name": "Oczyszczanie",
        "type": "cleaning",
        "url": "/categories/cleaning",
      },
      Object {
        "description": "Należy przestrzegać codziennej pielęgnacji twarzy",
        "id": "3",
        "name": "Pielęgnacja",
        "type": "care",
        "url": "/categories/care",
      },
    ]
  }
  fetchCategories={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <Nav
    categories={
      Array [
        Object {
          "description": "Profesjonalne SPA w domowym zaciszu to relaks dla ciała i duszy",
          "id": "1",
          "name": "Domowe SPA",
          "type": "homeSpa",
          "url": "/categories/home-spa",
        },
        Object {
          "description": "Oczyszczanie to podstawowy krok w dojściu do pięknej cery",
          "id": "2",
          "name": "Oczyszczanie",
          "type": "cleaning",
          "url": "/categories/cleaning",
        },
        Object {
          "description": "Należy przestrzegać codziennej pielęgnacji twarzy",
          "id": "3",
          "name": "Pielęgnacja",
          "type": "care",
          "url": "/categories/care",
        },
      ]
    }
  />
</Navigation>
`;

exports[`should render correctly without categories 1`] = `""`;
